spec/custom_tags_spec.rb

```diff
@@ -5,30 +5,30 @@ def fig( opts )
     label, img = opts[:text].split('|').map! {|str| str.strip}
 
     html  = %Q{<div class="img" id="figure-#{label.tr('.', '-')}">\n}
-    html << %Q{  <a class="fig" href="/images/#{img}">\n}
-    html << %Q{    <img src="/images/thumbs/#{img}" alt="Figure #{label}" />\n}
-    html << %Q{  </a>\n}
-    html << %Q{  <p>Figure #{label}</p>\n}
-    html << %Q{<div>\n}
+    html += %Q{  <a class="fig" href="/images/#{img}">\n}
```

Comment:
Better to make the initial string mutable (using `.dup` at the end), and then keeping all the `<<` calls. Using `+=` means extra string allocations that aren't really necessary.

Reply:
That makes sense. It makes the diff smaller too.